packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java

```diff
@@ -192,18 +192,25 @@ private Element doRender(VariableManager nodeVariableManager, String targetObjec
             parentState = ViewModifier.Hidden;
         }
 
+        var childNodeVariableManager = nodeVariableManager.createChild();
+        var oldAncestors = nodeVariableManager.get(VariableManager.ANCESTORS, List.class).orElse(List.of());
+        var ancestors = new ArrayList<>();
+        ancestors.addAll(oldAncestors);
+        nodeVariableManager.get(VariableManager.SELF, Object.class).ifPresent(ancestors::add);
+        childNodeVariableManager.put(VariableManager.ANCESTORS, ancestors);
+
         NodeChildrenComponentProps nodeChildrenComponentProps = NodeChildrenComponentProps.newNodeChildrenComponentProps()
                 .nodeComponentProps(this.props)
-                .variableManager(nodeVariableManager)
+                .variableManager(childNodeVariableManager)
                 .parentState(parentState)
                 .state(state)
                 .previousParentNode(optionalPreviousNode.orElse(null))
                 .diagramEvents(diagramEvents)
                 .build();
         Element nodeChildren = new Element(NodeChildrenComponent.class, nodeChildrenComponentProps);
 
-        Integer defaultWidth = nodeDescription.getDefaultWidthProvider().apply(nodeVariableManager);
-        Integer defaultHeight = nodeDescription.getDefaultHeightProvider().apply(nodeVariableManager);
+        Integer defaultWidth = nodeDescription.getDefaultWidthProvider().apply(childNodeVariableManager);
+        Integer defaultHeight = nodeDescription.getDefaultHeightProvider().apply(childNodeVariableManager);
```

Comment:
Don't change this, those providers / expressions are for the current node, not the child nodes

Reply:
Done